### DISCUSSION TITLE
Run project script on error

### DISCUSSION THREAD

**Discussion by andrewpbray** [quarto-dev/quarto-cli#9216] [Q&A] [answered]
### Description

I have a custom project type with a pre-render script that creates extra files and a post-render that cleans them up. The problem arises when there's an error during the render process: it doesn't trigger the post-render script and thus leaves the extra files in the directory. What is the best way to handle this? I'm sniffing around for an `on-error` option I guess.

**Comment by cscheid** [comment, marked as answer]
> What is the best way to handle this? I'm sniffing around for an `on-error` option I guess.

We don't really have one.
